src/bin/basic_subclass.rs

```diff
@@ -0,0 +1,238 @@
+// basic_subclass.rs
+//
+// Copyright 2019 Jordan Petridis <jpetridis@gnome.org>
```

Comment:
Well, I have a few issues on this but more globally, can you take a look at the other examples' header and do something similar please?